### DISCUSSION TITLE
Do I need to handle routing myself when using Portals (read-only replicas)?

### DISCUSSION THREAD

**Discussion by kristjanmar** [planetscale/discussion#400] [Q&A] [answered]
Hello, I am considering PlanetScale for my database needs, with a back-end running on Cloudflare Workers.

I am using the serverless driver and I am also interested in using the Portals feature to create read-only replicas since I get traffic from all over the world.

Is there a way to handle the routing to the closest read replica automatically? Or do I need to select different database URLs based on the location of the Worker?

Currently I am using MacroMeta which has geodistributed nodes, but the routing to the nearest node is handled automatically.

**Comment by orware** [comment, marked as answer]
Hello @kristjanmar,

Currently, you would need to handle the routing yourself using an approach similar to what is described in Fly.io's [Multi-Region Laravel with PlanetScale](https://fly.io/laravel-bytes/multi-region-laravel-with-planetscale/) article, although the approach might have to be tweaked for use with Cloudflare Workers as I don't have a resource that specifically covers the steps required to share with you.

I hope this is helpful!
